pkg/resources/validation/cluster/provisioningcluster.go

```diff
@@ -164,6 +178,93 @@ func (p *ProvisioningClusterValidator) validateClusterName(request *admission.Re
 	return nil
 }
 
+// validatePSACT validate if the cluster and underlying secret are configured properly when PSACT is enabled or disabled
+func (p *ProvisioningClusterValidator) validatePSACT(request *admission.Request, response *admissionv1.AdmissionResponse, cluster *v1.Cluster) error {
+	if cluster.Name == "local" || cluster.Spec.RKEConfig == nil {
+		return nil
+	}
+
+	name := fmt.Sprintf(mutator.SecretName, cluster.Name)
+	mountPath := fmt.Sprintf(mutator.MountPath, mutator.GetRuntime(cluster.Spec.KubernetesVersion))
+	templateName := cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+	switch request.Operation {
+	case admissionv1.Delete:
+		_, err := p.secretCache.Get(cluster.Namespace, name)
+		if err == nil {
+			return fmt.Errorf("[provsioning cluster validator] the secret %s still exists in the cluster", name)
+		} else if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("[provsioning cluster validator] failed to validate if the secret exists: %w", err)
+		} else {
+			return nil
+		}
```

Comment:
We don't need else with early returns.
```suggestion
		if err == nil {
			return fmt.Errorf("[provsioning cluster validator] the secret %s still exists in the cluster", name)
		} 
		if !apierrors.IsNotFound(err) {
			return fmt.Errorf("[provsioning cluster validator] failed to validate if the secret exists: %w", err)
		} 
		return nil
		
```

Reply:
fixed.